apps/files/models.py

```diff
@@ -139,53 +139,63 @@ def get_url_path(self, src):
     @classmethod
     def from_upload(cls, upload, version, platform, is_beta=False,
                     parse_data={}):
-        f = cls(version=version, platform=platform)
+        addon = version.addon
+
+        file_ = cls(version=version, platform=platform)
         upload.path = amo.utils.smart_path(nfd_str(upload.path))
         ext = os.path.splitext(upload.path)[1]
         if ext == '.jar':
             ext = '.xpi'
-        f.filename = f.generate_filename(extension=ext or '.xpi')
+        file_.filename = file_.generate_filename(extension=ext or '.xpi')
         # Size in bytes.
-        f.size = storage.size(upload.path)
+        file_.size = storage.size(upload.path)
         data = cls.get_jetpack_metadata(upload.path)
         if 'sdkVersion' in data and data['sdkVersion']:
-            f.jetpack_version = data['sdkVersion'][:10]
-        if f.jetpack_version:
-            Tag(tag_text='jetpack').save_tag(version.addon)
-        f.builder_version = data['builderVersion']
-        f.no_restart = parse_data.get('no_restart', False)
-        f.strict_compatibility = parse_data.get('strict_compatibility', False)
-        f.is_multi_package = parse_data.get('is_multi_package', False)
-        if version.addon.status == amo.STATUS_PUBLIC:
-            if is_beta:
-                f.status = amo.STATUS_BETA
-            elif version.addon.trusted:
-                f.status = amo.STATUS_PUBLIC
-        elif (version.addon.status in amo.LITE_STATUSES and
-              version.addon.trusted):
-            f.status = version.addon.status
-        f.hash = f.generate_hash(upload.path)
+            file_.jetpack_version = data['sdkVersion'][:10]
+        if file_.jetpack_version:
+            Tag(tag_text='jetpack').save_tag(addon)
+        file_.builder_version = data['builderVersion']
+        file_.no_restart = parse_data.get('no_restart', False)
+        file_.strict_compatibility = parse_data.get('strict_compatibility',
+                                                    False)
+        file_.is_multi_package = parse_data.get('is_multi_package', False)
+
+        if is_beta and addon.status == amo.STATUS_PUBLIC:
+            file_.status = amo.STATUS_BETA
+        elif addon.trusted:
+            # New files in trusted add-ons automatically receive the correct
+            # approved status for their review class.
```

Comment:
I'm not sure. This doesn't change any of the logic (the tests verify this). I just slightly reorganized the conditionals since, when I read this code to try to verify a comment, it took me three read-throughs to realize most of it only applied to trusted add-ons.